### ISSUE TITLE
Password still required to refresh or update

### ISSUE THREAD

**Issue by neogeographica** [elementary/appcenter#1365]
## Prerequisites
- [X] I have searched open and closed issues for duplicates.

## Describe the bug
App Center (3.4.0+r2468+pkg92~ubuntu5.1.4.1) required me to enter my login password in order to refresh:
![pwd_to_refresh](https://user-images.githubusercontent.com/2068148/84324317-5cfa3180-ab2d-11ea-97dc-ecf55cc45ea3.png)

And then to update the Code app:
![pwd_to_update](https://user-images.githubusercontent.com/2068148/84324362-6b484d80-ab2d-11ea-86b4-74f710ce9ebf.png)

And subsequently clicking Update for "Operating System Updates" asked for my password again.

Closing and re-opening App Center again asked me for my password to refresh.

FWIW, in "Passwords and Keys" (seahorse), my Login keyring is (and was, before opening App Center) currently unlocked, if that's at all relevant.

I'm willing to do additional tests & provide other info if that would be helpful.

### To Reproduce
1. Open the App Center. First symptom seen here; pwd required for refresh.
2. Click on 'Installed'.
3. Find an app that has a pending update; click on 'Update'. Second symptom seen here; pwd required for update.

### Expected behavior
From the latest blog on Hera updates, I understand that with the latest App Center, admin password should not be required to update already-installed packages. (I'm not 100% clear on whether it should also not be required for refresh, although that seems like a corollary.)

## Platform Information

![system_settings](https://user-images.githubusercontent.com/2068148/84324999-a8610f80-ab2e-11ea-824e-27f88c304c2c.png)

- [X] I'm using the latest released stable version

**Comment by davidmhewitt** [comment]
On a fresh elementary install, you wouldn't be prompted for authentication to refresh the repositories, even prior to the change where updates don't require a password. So there's something odd going on here.

Can you confirm that you're on a local session? i.e. you're sat at the computer in question using the keyboard and mouse? Not accessing it via RDP or VNC etc...

There's a security measure where if you're considered a "remote" user, you'll be asked for a password for more things.

**Comment by neogeographica** [comment]
> Can you confirm that you're on a local session? i.e. you're sat at the computer in question using the keyboard and mouse?

Yep.

**Comment by davidmhewitt** [comment]
In that case, can you run policykit with debugging enabled and then reproduce the issue by doing the following:

```
sudo killall polkitd
sudo G_MESSAGES_DEBUG=all /usr/lib/policykit-1/polkitd
```

Then in another tab, leaving that one running, run up AppCenter and reproduce the issue:
```
killall io.elementary.appcenter
io.elementary.appcenter
```

If you copy the output from your polkit tab, we may be able to see what's going on. Probably advisable to reboot your system after doing this too, so that polkit can go back to normal.

**Comment by davidmhewitt** [comment]
The output of the following command is probably also useful to us:

```
pkaction --action-id org.freedesktop.packagekit.system-update --verbose
```

**Comment by neogeographica** [comment]
Something restarts polkitd as soon as it is killed:
```
joel@lazarus:~/tmp$ ps -efl | grep polkitd
4 S root      1067     1  0  80   0 - 73199 -      09:09 ?        00:00:00 /usr/lib/policykit-1/polkitd --no-debug
0 S joel      4359  4287  0  80   0 - 10011 pipe_w 09:09 pts/0    00:00:00 grep --color=auto polkitd

joel@lazarus:~/tmp$ sudo killall polkitd
[sudo] password for joel:

joel@lazarus:~/tmp$ ps -efl | grep polkitd
4 S root      4427     1 15  80   0 - 73201 -      09:09 ?        00:00:00 /usr/lib/policykit-1/polkitd --no-debug
0 S joel      4465  4287  0  80   0 - 10011 pipe_w 09:09 pts/0    00:00:00 grep --color=auto polkitd

```

**Comment by davidmhewitt** [comment]
> Something restarts polkitd as soon as it is killed:

Ah, you may be able to start it yourself before it restarts by doing both commands in one, like:
```
sudo killall polkitd && sudo G_MESSAGES_DEBUG=all /usr/lib/policykit-1/polkitd
```

See if that allows you to get the debugging output from polkit.

The output of your `pkaction` command looks normal, so nothing has modified that config. It may also be useful to know what groups your account is a member of, which you can check by running `groups`

It's hard to know what has caused this really, I guess it's possible that a package you installed messed with some config, but I don't think it's very likely.

**Comment by neogeographica** [comment]
I'll attach the entire debug output as a txt file at the end of this, but here's some snippets:

polkitd debug output during the time after starting App Center and before the Authentication Required dialog appears:
```
** (polkitd:9540): DEBUG: 08:25:30.081: system-bus-name::1.230 is inquiring whether system-bus-name::1.50 is authorized for org.freedesktop.login1.inhibit-handle-lid-switch
** (polkitd:9540): DEBUG: 08:25:30.081:  user of caller is unix-user:root
** (polkitd:9540): DEBUG: 08:25:30.082:  user of subject is unix-user:joel
** (polkitd:9540): DEBUG: 08:25:30.082: checking whether system-bus-name::1.50 is authorized for org.freedesktop.login1.inhibit-handle-lid-switch
** (polkitd:9540): DEBUG: 08:25:30.083:   0x7fe084017090
** (polkitd:9540): DEBUG: 08:25:30.083: Checking whether session c2 is active.
** (polkitd:9540): DEBUG: 08:25:30.083: Session c2 has UID 1000.
** (polkitd:9540): DEBUG: 08:25:30.083: UID 1000 has state online.
** (polkitd:9540): DEBUG: 08:25:30.083:  subject is in session c2 (local=1 active=0)
** (polkitd:9540): DEBUG: 08:25:30.083:  is authorized (has implicit authorization local=1 active=0)
** (polkitd:9540): DEBUG: 08:25:30.083:  
** (polkitd:9540): DEBUG: 08:25:33.771: system-bus-name::1.254 is inquiring whether system-bus-name::1.253 is authorized for org.freedesktop.packagekit.system-sources-refresh
** (polkitd:9540): DEBUG: 08:25:33.772:  user of caller is unix-user:root
** (polkitd:9540): DEBUG: 08:25:33.772:  user of subject is unix-user:joel
** (polkitd:9540): DEBUG: 08:25:33.772: checking whether system-bus-name::1.253 is authorized for org.freedesktop.packagekit.system-sources-refresh
** (polkitd:9540): DEBUG: 08:25:33.773:   0x7fe084050a00
** (polkitd:9540): DEBUG: 08:25:33.773: Checking whether session c2 is active.
** (polkitd:9540): DEBUG: 08:25:33.773: Session c2 has UID 1000.
** (polkitd:9540): DEBUG: 08:25:33.773: UID 1000 has state online.
** (polkitd:9540): DEBUG: 08:25:33.773:  subject is in session c2 (local=1 active=0)
** (polkitd:9540): DEBUG: 08:25:33.774:  challenge (implicit_authorization = auth_admin)
** (polkitd:9540): DEBUG: 08:25:33.775:  
** (polkitd:9540): DEBUG: 08:25:33.776:  using authentication agent for challenge

** (polkitd:9540): WARNING **: 08:25:33.778: Error parsing identity unix-group:admin: No UNIX group with name admin: Success
```

After entering my password there:
```
** (polkitd:9540): DEBUG: 08:26:42.037: In authentication_agent_response for cookie '1-c45c2f6cf21bfaf62fce501fe17e06d5-1-c6337f87536fcd1d00baea1ed9645b9d' and identity unix-user:joel
** (polkitd:9540): DEBUG: 08:26:42.041: Authentication complete, is_authenticated = 1
** (polkitd:9540): DEBUG: 08:26:42.041: In check_authorization_challenge_cb
  subject                system-bus-name::1.253
  action_id              org.freedesktop.packagekit.system-sources-refresh
  was_dismissed          0
  authentication_success 1
```

After a while with App Center sitting on the Installed tab, more output:
```
** (polkitd:9540): DEBUG: 08:27:25.849: system-bus-name::1.230 is inquiring whether system-bus-name::1.50 is authorized for org.freedesktop.login1.inhibit-handle-lid-switch
** (polkitd:9540): DEBUG: 08:27:25.850:  user of caller is unix-user:root
** (polkitd:9540): DEBUG: 08:27:25.851:  user of subject is unix-user:joel
** (polkitd:9540): DEBUG: 08:27:25.851: checking whether system-bus-name::1.50 is authorized for org.freedesktop.login1.inhibit-handle-lid-switch
** (polkitd:9540): DEBUG: 08:27:25.852:   0x7fe08404f690
** (polkitd:9540): DEBUG: 08:27:25.852: Checking whether session c2 is active.
** (polkitd:9540): DEBUG: 08:27:25.852: Session c2 has UID 1000.
** (polkitd:9540): DEBUG: 08:27:25.852: UID 1000 has state online.
** (polkitd:9540): DEBUG: 08:27:25.852:  subject is in session c2 (local=1 active=0)
** (polkitd:9540): DEBUG: 08:27:25.853:  is authorized (has implicit authorization local=1 active=0)
** (polkitd:9540): DEBUG: 08:27:25.853:  
```


No packages need updating at the moment so it's not super easy to test that part, but I can try to arrange it if necessary.

[polkit-debug.txt](https://github.com/elementary/appcenter/files/4771710/polkit-debug.txt)

**Comment by davidmhewitt** [comment]
So the problem here is that polkit thinks your session is inactive as seen by `active=0` throughout that file.

Could you run the following and post the output?
```
loginctl session-status c2
````

**Comment by neogeographica** [comment]
```
c2 - joel (1000)
           Since: Fri 2020-06-12 08:37:36 PDT; 1h 11min ago
          Leader: 2470 (lightdm)
            Seat: seat0; vc7
         Display: :0
         Service: lightdm; type x11; class user
         Desktop: pantheon
           State: opening
            Unit: session-c2.scope
                  ├─2470 lightdm --session-child 13 20
                  ├─3548 /usr/bin/gnome-keyring-daemon --daemonize --login
                  ├─3551 /usr/lib/gnome-session/gnome-session-binary --session=pantheon
                  ├─3680 /usr/bin/ssh-agent /usr/bin/im-launch gnome-session --session=pantheon
                  ├─3707 /usr/lib/gnome-settings-daemon/gsd-print-notifications
                  ├─3711 /usr/lib/gnome-settings-daemon/gsd-a11y-settings
                  ├─3714 /usr/lib/gnome-settings-daemon/gsd-smartcard
                  ├─3721 /usr/lib/gnome-settings-daemon/gsd-wacom
                  ├─3725 /usr/lib/gnome-settings-daemon/gsd-power
                  ├─3726 /usr/lib/gnome-settings-daemon/gsd-housekeeping
                  ├─3727 /usr/lib/gnome-settings-daemon/gsd-xsettings
                  ├─3728 /usr/lib/gnome-settings-daemon/gsd-sharing
                  ├─3731 /usr/lib/gnome-settings-daemon/gsd-media-keys
                  ├─3739 /usr/lib/gnome-settings-daemon/gsd-sound
                  ├─3743 /usr/lib/gnome-settings-daemon/gsd-mouse
                  ├─3757 /usr/lib/gnome-settings-daemon/gsd-clipboard
                  ├─3762 /usr/lib/gnome-settings-daemon/gsd-keyboard
                  ├─3774 /usr/lib/gnome-settings-daemon/gsd-color
                  ├─3776 /usr/lib/gnome-settings-daemon/gsd-datetime
                  ├─3781 /usr/bin/pulseaudio --start --log-target=syslog
                  ├─3789 /usr/lib/gnome-settings-daemon/gsd-rfkill
                  ├─3806 /usr/lib/gnome-settings-daemon/gsd-printer
                  ├─3894 gala
                  ├─3910 plank
                  ├─3912 wingpanel
                  ├─3913 /usr/lib/x86_64-linux-gnu/io.elementary.cerbere
                  ├─3986 gala-daemon
                  ├─3994 /usr/lib/x86_64-linux-gnu/indicator-application/indicator-application-service
                  ├─3998 albert
                  ├─4000 /home/joel/.wine/drive_c/Program Files (x86)/1Password 4/Agile1pAgent.exe
                  ├─4001 /usr/lib/x86_64-linux-gnu/geoclue2-1-pantheon/pantheon-agent-geoclue2
                  ├─4002 /usr/lib/x86_64-linux-gnu/pantheon-parental-controls-client
                  ├─4006 nm-applet
                  ├─4012 /usr/lib/x86_64-linux-gnu/policykit-1-pantheon/io.elementary.desktop.agent-polkit
                  ├─4017 /usr/lib/x86_64-linux-gnu/io.elementary.calendar-daemon
                  ├─4020 light-locker
                  ├─4035 /opt/wine-stable/bin/wineserver
                  ├─4064 /home/joel/.dropbox-dist/dropbox-lnx.x86_64-99.4.501/dropbox
                  ├─4108 C:\windows\system32\services.exe
                  ├─4136 C:\windows\system32\plugplay.exe
                  ├─4152 C:\windows\system32\winedevice.exe
                  ├─4247 C:\windows\system32\explorer.exe /desktop
                  ├─4267 C:\windows\system32\winedevice.exe
                  ├─4649 io.elementary.appcenter -s
                  ├─4677 sh -c /usr/lib/x86_64-linux-gnu/libproxy/0.4.14/pxgsettings org.gnome.system.proxy org.gnome.system.proxy.http org
                  ├─4678 /usr/lib/x86_64-linux-gnu/libproxy/0.4.14/pxgsettings org.gnome.system.proxy org.gnome.system.proxy.http org.gnome
                  ├─5083 /usr/share/discord/Discord
                  ├─5087 /usr/share/discord/Discord --type=zygote
                  ├─5089 /usr/share/discord/Discord --type=zygote
                  ├─5173 /usr/share/discord/Discord --type=gpu-process --field-trial-handle=16760364927845385530,9685717123633436832,131072
                  ├─5182 /usr/share/discord/Discord --type=utility --field-trial-handle=16760364927845385530,9685717123633436832,131072 --d
                  ├─5210 /usr/share/discord/Discord --type=broker
                  ├─5239 /usr/share/discord/Discord --type=renderer --autoplay-policy=no-user-gesture-required --field-trial-handle=1676036
                  ├─6574 /usr/share/discord/Discord --type=utility --field-trial-handle=16760364927845385530,9685717123633436832,131072 --d
                  ├─7590 io.elementary.terminal
                  ├─7599 /bin/bash
                  ├─7710 /bin/bash
                  ├─8352 loginctl session-status c2
                  └─8353 pager

Jun 12 09:48:56 lazarus plank.desktop[3910]: [DEBUG 09:48:56.324557] [GdkPixbuf]         Dimensions: 16 x 16
Jun 12 09:49:11 lazarus gnome-keyring-daemon[3548]: asked to register item /org/freedesktop/secrets/collection/login/1, but it's already re
Jun 12 09:49:12 lazarus gnome-keyring-daemon[3548]: asked to register item /org/freedesktop/secrets/collection/login/1, but it's already re
Jun 12 09:49:12 lazarus plank.desktop[3910]: [7794:7806:0612/094912.103333:ERROR:nss_util.cc(283)] After loading Root Certs, loaded==false:
Jun 12 09:49:13 lazarus indicator-appli[3994]: Notification Item on object /StatusNotifierItem of org.kde.StatusNotifierItem-7753-2 doesn't
Jun 12 09:49:16 lazarus plank.desktop[3910]: Fontconfig error: Cannot load default config file: No such file: (null)
Jun 12 09:49:16 lazarus plank.desktop[3910]: Fontconfig error: Cannot load default config file: No such file: (null)
Jun 12 09:49:19 lazarus plank.desktop[3910]: [1,1084168128:16:49:19.983594] Native Client module will be loaded at base address 0x00004d250
Jun 12 09:49:21 lazarus plank.desktop[3910]: [0612/094921.749176:ERROR:nacl_helper_linux.cc(308)] NaCl helper process running without a san
Jun 12 09:49:21 lazarus plank.desktop[3910]: Most likely you need to configure your SUID sandbox correctly
```

**Comment by neogeographica** [comment]
Also trying it immediately after reboot and login:
```
c2 - joel (1000)
           Since: Fri 2020-06-12 09:56:33 PDT; 23s ago
          Leader: 2153 (lightdm)
            Seat: seat0; vc7
         Display: :0
         Service: lightdm; type x11; class user
         Desktop: pantheon
           State: active
            Unit: session-c2.scope
                  ├─2153 lightdm --session-child 13 20
                  ├─3282 /usr/bin/gnome-keyring-daemon --daemonize --login
                  ├─3285 /usr/lib/gnome-session/gnome-session-binary --session=pantheon
                  ├─3414 /usr/bin/ssh-agent /usr/bin/im-launch gnome-session --session=pantheon
                  ├─3441 /usr/lib/gnome-settings-daemon/gsd-print-notifications
                  ├─3445 /usr/lib/gnome-settings-daemon/gsd-a11y-settings
                  ├─3446 /usr/lib/gnome-settings-daemon/gsd-smartcard
                  ├─3455 /usr/lib/gnome-settings-daemon/gsd-wacom
                  ├─3458 /usr/lib/gnome-settings-daemon/gsd-power
                  ├─3460 /usr/lib/gnome-settings-daemon/gsd-housekeeping
                  ├─3461 /usr/lib/gnome-settings-daemon/gsd-xsettings
                  ├─3465 /usr/lib/gnome-settings-daemon/gsd-sharing
                  ├─3466 /usr/lib/gnome-settings-daemon/gsd-media-keys
                  ├─3469 /usr/lib/gnome-settings-daemon/gsd-sound
                  ├─3476 /usr/lib/gnome-settings-daemon/gsd-mouse
                  ├─3496 /usr/lib/gnome-settings-daemon/gsd-clipboard
                  ├─3503 /usr/lib/gnome-settings-daemon/gsd-keyboard
                  ├─3508 /usr/lib/gnome-settings-daemon/gsd-color
                  ├─3512 /usr/bin/pulseaudio --start --log-target=syslog
                  ├─3513 /usr/lib/gnome-settings-daemon/gsd-datetime
                  ├─3520 /usr/lib/gnome-settings-daemon/gsd-rfkill
                  ├─3548 /usr/lib/gnome-settings-daemon/gsd-printer
                  ├─3614 gala
                  ├─3629 plank
                  ├─3631 wingpanel
                  ├─3632 /usr/lib/x86_64-linux-gnu/io.elementary.cerbere
                  ├─3693 gala-daemon
                  ├─3713 /usr/lib/x86_64-linux-gnu/indicator-application/indicator-application-service
                  ├─3716 albert
                  ├─3718 /home/joel/.wine/drive_c/Program Files (x86)/1Password 4/Agile1pAgent.exe
                  ├─3720 /usr/lib/x86_64-linux-gnu/geoclue2-1-pantheon/pantheon-agent-geoclue2
                  ├─3721 /usr/lib/x86_64-linux-gnu/pantheon-parental-controls-client
                  ├─3722 nm-applet
                  ├─3731 /usr/lib/x86_64-linux-gnu/policykit-1-pantheon/io.elementary.desktop.agent-polkit
                  ├─3736 /usr/lib/x86_64-linux-gnu/io.elementary.calendar-daemon
                  ├─3739 light-locker
                  ├─3773 /opt/wine-stable/bin/wineserver
                  ├─3783 /home/joel/.dropbox-dist/dropbox-lnx.x86_64-99.4.501/dropbox
                  ├─3824 C:\windows\system32\services.exe
                  ├─3849 C:\windows\system32\plugplay.exe
                  ├─3881 C:\windows\system32\winedevice.exe
                  ├─3979 C:\windows\system32\explorer.exe /desktop
                  ├─3999 C:\windows\system32\winedevice.exe
                  ├─4397 io.elementary.terminal
                  ├─4405 /bin/bash
                  ├─4434 loginctl session-status c2
                  └─4435 pager
```

In that case it says "active". And App Center refreshes without the password prompt. I'll experiment a bit more.

**Comment by neogeographica** [comment]
Hmm, so FWIW:

* After reboot: session is marked "active".
* Allow desktop to lock on idle then log back in: session is marked "active".
* Launch a bunch of apps, allow desktop to lock on idle, log back in: session is marked "opening".

Apps in this test were Chrome, Discord, and Sublime Text. Might be a coincidence?

I'll pause here.

**Comment by davidmhewitt** [comment]
I think you’d probably want to look in syslog around the time of unlocking the session to see if there was anything that crashed or any other errors that would make logind think that the session failed to resume properly.

At this point though, I’m sure that this isn’t a problem with AppCenter, so I’m going to close the issue. Feel free to keep commenting on the thread with your findings and I’ll help out if/when I can.

**Comment by neogeographica** [comment]
All righty. Thanks for the pointers!

**Comment by neogeographica** [comment]
So... a long time ago when first starting out with elementary, I had to do this to keep the system from suspending:
```sudo systemctl mask sleep.target suspend.target hibernate.target hybrid-sleep.target```
as it would not honor my "suspend never" choice in the power settings.

I still had that masking in place, as the problem had not been fixed last time I checked.

I tried doing this now, to undo that change:
```sudo systemctl unmask sleep.target suspend.target hibernate.target hybrid-sleep.target```
and that seems to have fixed the session problem. Not 100% sure yet, but it has been a couple of days with no issues so far.

Also the suspend bug seems to have been fixed, so all good perhaps.
